src/confluent_kafka/src/Admin.c

```diff
@@ -976,6 +976,537 @@ static PyObject *Admin_alter_configs (Handle *self, PyObject *args,
+        cfl_PyDict_SetInt(kwargs, "operation",
+                                     rd_kafka_AclBinding_operation(c_acl_binding));
+        cfl_PyDict_SetInt(kwargs, "permission_type",
+                                     rd_kafka_AclBinding_permission_type(c_acl_binding));
+
+        args = PyTuple_New(0);
+        acl_binding = PyObject_Call(AclBinding_type, args, kwargs);
+
+        Py_DECREF(args);
+        Py_DECREF(kwargs);
+        Py_DECREF(AclBinding_type);
+        return acl_binding;
+}
+
+/**
+ * @brief Convert C AclBinding array to py list.
+ */
+static PyObject *
+Admin_c_AclBindings_to_py (const rd_kafka_AclBinding_t **c_acls,
+                                          size_t c_acls_cnt) {
+        size_t i;
+        PyObject *result;
+
+        result = PyList_New(c_acls_cnt);
+
+        for (i = 0 ; i < c_acls_cnt ; i++) {
+                PyList_SET_ITEM(result, i,
+                        Admin_c_AclBinding_to_py(c_acls[i]));
+        }
+
+        return result;
+}
+
+
+/**
+ * @brief Convert C acl_result_t array to py list.
+ */
+static PyObject *
+Admin_c_acl_result_to_py (const rd_kafka_acl_result_t **c_result,
+                            size_t cnt) {
+        PyObject *result;
+        size_t ti;
+
+        result = PyList_New(cnt);
+
+        for (ti = 0 ; ti < cnt ; ti++) {
+                PyObject *error;
+                const rd_kafka_error_t *c_error = rd_kafka_acl_result_error(c_result[ti]);
+
+                error = KafkaError_new_or_None(
+                        rd_kafka_error_code(c_error),
+                        rd_kafka_error_name(c_error));
+
+                PyList_SET_ITEM(result, ti, error);
+        }
+
+        return result;
+}
+
+
+/**
+ * @brief Convert C DeleteAcls result response array to py list.
+ */
+static PyObject *
+Admin_c_DeleteAcls_result_responses_to_py (const rd_kafka_DeleteAcls_result_response_t **c_result_responses,
+                            size_t cnt) {
+        const rd_kafka_AclBinding_t **c_matching_acls;
+        size_t c_matching_acls_cnt;
+        PyObject *result;
+        size_t ti;
+
+        result = PyList_New(cnt);
+
+        for (ti = 0 ; ti < cnt ; ti++) {
+                PyObject *error;
+                const rd_kafka_error_t *c_error = rd_kafka_DeleteAcls_result_response_error(c_result_responses[ti]);
+
+                if (c_error) {
+                        error = KafkaError_new_or_None(
+                                rd_kafka_error_code(c_error),
+                                rd_kafka_error_name(c_error));
+                        PyList_SET_ITEM(result, ti, error);
+                } else {
+                        c_matching_acls = rd_kafka_DeleteAcls_result_response_matching_acls(
+                                                                        c_result_responses[ti],
+                                                                        &c_matching_acls_cnt);
+                        PyList_SET_ITEM(result, ti,
```

Comment:
So here we're mixing two types in the same result list; KafkaErrors and AclBindings. Is that intended? Seems like it makes it harder to use.

Reply:
It's intended, this way _make_acls_result can be used in create_acls and delete_acls, just in create_acls it's KafkaError or None, here is KafkaError or AclBinding. Also _make_acls_result is similiar to _make_topics_result. If it's better to change it, it could be converted into a tuple(result, error) in both create_acls and delete_acls.

Reply:
Huhm, I can't quite visualize the usage.
Could you show an example of what error handling of these results would look like on the application (or test) side with the current approach?

Reply:
the recipient of this is a private method [_make_acls_result](https://github.com/confluentinc/confluent-kafka-python/blob/83669187e0980343e6c454ce3ac74484916be152/src/confluent_kafka/admin/__init__.py#L132), what the user sees  instead is a future that either has a result with a list of `AclBinding` or an exception of type `KafkaException` wrapping the `KafkaError`.
An example of the usage is in the [tests](https://github.com/confluentinc/confluent-kafka-python/blob/83669187e0980343e6c454ce3ac74484916be152/tests/integration/admin/test_basic_operations.py#L65)